Rename scripts directory to bin

MauricioFauth opened the pull request:
- Related to #18512

MauricioFauth committed:
Rename scripts directory to bin

- Related to #18512

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

williamdes reviewed (approved):
Can we have bin/internal/ for scripts that are only internal and not needed in the final vendor bundles ?

williamdes reviewed (approved):
Can we have bin/internal/ for scripts that are only internal and not needed in the final vendor bundles ?

MauricioFauth commented:
> Can we have bin/internal/ for scripts that are only internal and not needed in the final vendor bundles ?

Aren't all scripts internal? We remove this directory when creating a release.

williamdes commented:
> > Can we have bin/internal/ for scripts that are only internal and not needed in the final vendor bundles ?
> 
> Aren't all scripts internal? We remove this directory when creating a release.

No they are not
And we should not remove the directory on some type of bundles, the source one I think
That said such bundles do not need internal scripts like check release excludes or create release

williamdes commented on bin/lang-cleanup.sh:
Internal

williamdes commented on bin/locales-contributors:
internal

williamdes commented on bin/set-version.sh:
internal

williamdes commented on bin/upload-release:
very internal

williamdes commented on bin/transformations_generator_plugin.sh:
internal

williamdes commented on bin/check-release-excludes.sh:
very internal

MauricioFauth commented:
> No they are not
> And we should not remove the directory on some type of bundles, the source one I think
> That said such bundles do not need internal scripts like check release excludes or create release

Besides the transformations scripts (which should be removed in my opinion, and replaced by proper interfaces), all scripts are internal. That's why I don't see the point of having the internal directory.

I changed the create release script to not remove the bin directory for the source bundle. Should I also do that for 5.2?

williamdes commented:
> > No they are not
> > And we should not remove the directory on some type of bundles, the source one I think
> > That said such bundles do not need internal scripts like check release excludes or create release
> 
> Besides the transformations scripts (which should be removed in my opinion, and replaced by proper interfaces), all scripts are internal. That's why I don't see the point of having the internal directory.

Okay. For example in https://salsa.debian.org/phpmyadmin-team/phpmyadmin if I had the generate mo script it could re build the mo files from source too

> I changed the create release script to not remove the bin directory for the source bundle. Should I also do that for 5.2?

no, that's okay

MauricioFauth committed:
Move some scripts to bin/internal

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>
